### PR TITLE
[enhancement](flush) make writer write fail status visible

### PR DESCRIPTION
**Pull request by Hastyshell**
## Proposed changes

When flushing memtable  failed, msg is invisible now, which is confused when locating problems.

In this PR:
1. Recording flush status with msg rather than a single status code during memtable's async flushing, to expose real status to fe and client.
2. Log out total disk capacity when the data dir has no capacity.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run click bench

**Comment by TangSiyang2001** [comment]
run click bench

**Comment by TangSiyang2001** [comment]
run click bench

**Comment by TangSiyang2001** [comment]
run clickbench

File: be/src/olap/rowset/beta_rowset_writer.cpp [hunk 1]
```diff
@@ -823,7 +823,7 @@ Status BetaRowsetWriter::_flush_segment_writer(std::unique_ptr<segment_v2::Segme
     Status s = (*writer)->finalize(&segment_size, &index_size);
     if (!s.ok()) {
         LOG(WARNING) << "failed to finalize segment: " << s.to_string();
-        return Status::Error<WRITER_DATA_WRITE_ERROR>();
+        return Status::Error<WRITER_DATA_WRITE_ERROR>(s.to_string());
```
Comment (dataroaring) [inline be/src/olap/rowset/beta_rowset_writer.cpp:826]: keep the error code.

**Review by dataroaring — CHANGES_REQUESTED** [review]
LGTM

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run p0

**Comment by TangSiyang2001** [comment]
run build all

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run clickbench

**Comment by TangSiyang2001** [comment]
run buildall

**Review by dataroaring — APPROVED** [review]
LGTM

**Review by freemandealer — APPROVED** [review]
LGTM

**Pull request merged**